src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs

```diff
@@ -134,6 +134,14 @@ internal sealed partial class ConsoleHost
             }
 #endif
 
+            // put PSHOME in front of PATH so that calling `powershell` within `powershell` always starts the same running version
```

Comment:
1. Is this right for other applications?
2. I would prefer more deterministic `[System.Diagnostics.Process]::GetCurrentProcess().MainModule.FileName`